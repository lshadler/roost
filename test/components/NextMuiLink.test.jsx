import React from 'react';
import {render} from '@testing-library/react';

import NextMuiLink from '../../components/NextMuiLink';

jest.mock('next/router', () => ({
  useRouter: jest.fn(() => ({pathname: '/test'})),
}));

jest.mock('clsx', () => () => {});

describe('NextMuiLink Component', () => {
  test('renders in default state', () => {
    const {asFragment} = render(<NextMuiLink href="/test"/>);

    expect(asFragment()).toMatchSnapshot();
  });

  test('renders in naked state', () => {
    const {asFragment} = render(<NextMuiLink href="/test" naked={true}/>);

    expect(asFragment()).toMatchSnapshot();
  });

  test('renders with href object', () => {
    const href = {
      pathname: '/test',
    };
    const {asFragment} = render(<NextMuiLink href={href} naked={true}/>);

    expect(asFragment()).toMatchSnapshot();
  });
});
