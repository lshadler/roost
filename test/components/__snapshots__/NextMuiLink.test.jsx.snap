// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NextMuiLink Component renders in default state 1`] = `
<DocumentFragment>
  <a
    href="/test"
  />
</DocumentFragment>
`;

exports[`NextMuiLink Component renders in naked state 1`] = `
<DocumentFragment>
  <a
    href="/test"
  />
</DocumentFragment>
`;

exports[`NextMuiLink Component renders with href object 1`] = `
<DocumentFragment>
  <a
    href="/test"
  />
</DocumentFragment>
`;
